### PR TITLE
RecipeApp Production

### PR DESCRIPTION
**Pull request by alanluqman**
## Description
In this PR, I and my partner built this web app with all requirement that given.

### Project requirements
- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
Should be built with Devise.
- Food list:
    - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

### Screen spanshoots
log in
sign up
home
foods
create new food
recipe
create new recipe
recipe show
public recipe
add ingredient
update ingredient

edit profile

### CODE REVIEW

File: app/models/recipe.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Recipe < ApplicationRecord
```
Comment (muneebulrehman) [inline app/models/recipe.rb:1]: As you can see from the ER diagram, the recipe and food are connected to each other through the recipe_food. Kindly do so as setting up the associations between models in a correct way is a part of the core requirements. ⬇️
image

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        if user_signed_in?
+          render json: @recipes
+        else
+          render html: 'you are not allowed to see this content.'
+        end
+      end
+    end
+  end
+
+  def public_recipes
```
Comment (muneebulrehman) [inline app/controllers/recipes_controller.rb:16]: All the public recipes should have the information of the **total price** and **total food items** associated with them which is missing in your case. You can refer to [this video](https://www.loom.com/share/4e90dd1f3531432689584a4b4d9e5c53) where I demonstrated the problem in detail. Kindly fix this issue as this is a part of the core requirements for this project. Below is the screenshot where you can see how to implement the features. ⬇️
image

File: spec/requests/foods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      visit new_user_session_path
+      fill_in 'Email', with: 'test@gmail.com'
+      fill_in 'Password', with: 'password'
+      click_button 'Log in'
+      visit user_foods_path(user_id: @user.id)
+    end
+
+    it 'should display all the food details' do
+      expect(page).to have_content 'You need to sign in or sign up before continuing'
+    end
```
Comment (muneebulrehman) [inline spec/requests/foods_spec.rb:18]: - Optional. 
Kindly do the proper testing for this controller. Currently, you are not doing any tests related to the food controller.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Nice UI

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly add the tests for the `recipe_food_controller` and `recipe_controller`. These tests are missing in your case. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alanluqman** [comment]
Thanks @muneebulrehman for your review, I did all you required. I hope next reviewer could approve it as soon .

**Review by EricMbouwe — APPROVED** [review]
Hi @alanluqman, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔
- Stunning UI ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alanluqman** [comment]
thank you so much bro @Eric

**Pull request merged**